requests/src/requests/Requester.scala

```diff
@@ -204,43 +200,37 @@ case class Requester(verb: String,
         }
 
         connection.setInstanceFollowRedirects(false)
-        val upperCaseVerb = verb.toUpperCase
-        if (Requester.officialHttpMethods.contains(upperCaseVerb)) {
-          connection.setRequestMethod(upperCaseVerb)
+        if (Requester.officialHttpMethods.contains(CMD)) {
+          connection.setRequestMethod(CMD)  
         } else {
           // HttpURLConnection enforces a list of official http METHODs, but not everyone abides by the spec
           // this hack allows us set an unofficial http method
           connection match {
             case cs: HttpsURLConnection =>
               cs.getClass.getDeclaredFields.find(_.getName == "delegate").foreach{ del =>
                 del.setAccessible(true)
-                Requester.methodField.set(del.get(cs), upperCaseVerb)
+                Requester.methodField.set(del.get(cs), CMD)
               }
             case c =>
-              Requester.methodField.set(c, upperCaseVerb)
-          }
+              Requester.methodField.set(c, CMD)
+          }            
         }
 
-        for((k, v) <- blobHeaders) connection.setRequestProperty(k, v)
-
-        for((k, v) <- sess.headers) connection.setRequestProperty(k, v)
-
-        for((k, v) <- headers) connection.setRequestProperty(k, v)
-
-        for((k, v) <- compress.headers) connection.setRequestProperty(k, v)
+        Seq(blobHeaders, sess.headers, headers, compress.headers).
+           foreach(h => for((k, v) <- h) connection.setRequestProperty(k, v))
```

Comment:
This creates an extra `Seq` that the old code was avoiding.
 I would rollback it, also doesn't help fixing the bug you found.

Reply:
Yeah, just a thought